### PR TITLE
PP-671/custom-smart-wallet-factory-migration

### PR DESCRIPTION
**Pull request by franciscotobar**
## What

- Migrate CustomSmartWalletFactory unit testing

## Why

- Part of the web3 to ethers migration

## Refs

- [PP-671](https://rsklabs.atlassian.net/browse/PP-671)

### CODE REVIEW

File: test/factory/customSmartWalletFactory.test.ts [hunk 1]
```diff
@@ -0,0 +1,537 @@
...
+        );
+
+        const finalWorkerBalance = await token.balanceOf(worker.address);
+
+        expect(finalWorkerBalance).to.be.equal(initialWorkerBalance);
+        await expect(smartWallet.isInitialized()).to.eventually.be.true;
+      });
+
+      it('should initialize the smart wallet in the expected address paying fee', async function () {
+        const amountToPay = 500;
```
Comment (AndresQuijano) [inline test/factory/customSmartWalletFactory.test.ts:274]: I prefer using big values to be more realistic.

File: test/factory/customSmartWalletFactory.test.ts [hunk 2]
```diff
@@ -0,0 +1,537 @@
...
+              signature
+            )
+        ).to.be.rejectedWith('Unable to initialize SW');
+
+        const finalWorkerBalance = await token.balanceOf(worker.address);
+
+        expect(finalWorkerBalance).to.be.equal(initialWorkerBalance);
+      });
+
+      it('should fail when owner does not have funds to pay', async function () {
```
Comment (AndresQuijano) [inline test/factory/customSmartWalletFactory.test.ts:365]: Why the owner doesn't have funds here but it has in the previous test. I can't find where are you funding it.

File: test/factory/customSmartWalletFactory.test.ts [hunk 3]
```diff
@@ -0,0 +1,537 @@
...
+              signature
+            )
+        ).to.be.rejectedWith('Unable to initialize SW');
+
+        const finalWorkerBalance = await token.balanceOf(worker.address);
+
+        expect(finalWorkerBalance).to.be.equal(initialWorkerBalance);
+      });
+
+      it('should fail when invalid caller(Not relayHub)', async function () {
```
Comment (AndresQuijano) [inline test/factory/customSmartWalletFactory.test.ts:408]: To make it clear, I would create a wrongRelayHub instead of using one of the existing accounts.

**Review by AndresQuijano — COMMENTED** [review]
Minor comments.

File: test/factory/customSmartWalletFactory.test.ts [hunk 4]
```diff
@@ -0,0 +1,537 @@
...
+
+      it('should initialize the smart wallet in the expected address paying fee', async function () {
+        const amountToPay = 500;
+
+        const deployRequest = createDeployRequest(
+          {
+            from: owner.address,
+            tokenContract: token.address,
+            tokenAmount: amountToPay,
+            tokenGas: 55000,
```
[the remark this answers is not in the document]
Comment (franciscotobar) [reply test/factory/customSmartWalletFactory.test.ts:281]: updated

File: test/factory/customSmartWalletFactory.test.ts (hunk 1 above)
Comment (franciscotobar) [reply test/factory/customSmartWalletFactory.test.ts:274]: updated

File: test/factory/customSmartWalletFactory.test.ts (hunk 2 above)
Comment (franciscotobar) [reply test/factory/customSmartWalletFactory.test.ts:365]: updated

File: test/factory/customSmartWalletFactory.test.ts (hunk 3 above)
Comment (franciscotobar) [reply test/factory/customSmartWalletFactory.test.ts:408]: updated

**Review by AndresQuijano — APPROVED** [review]

**Pull request merged**